ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts

```diff
@@ -189,52 +187,51 @@ export class ActivityComponent extends BaseActivityComponent implements OnInit,
     public incrementStep(scroll: boolean = true): void {
         const nextIndex = this.nextAvailableSectionIndex();
         if (nextIndex !== -1) {
-            // enable any validation errors to be visible
-            this.validationRequested = true;
-            this.sendSectionAnalytics();
-            this.currentSection.validate();
-
             if (scroll) {
                 this.scrollToTop();
             }
-
-            if (this.currentSection.valid) {
-                this.resetValidationState();
-
-                this.isPageBusy.pipe(
-                    tap(_ => this.isLoaded$.next(false)),
-                    debounceTime(this.timeToDebounce),
-                    filter(busy => !busy),
-                    take(1)
-                ).subscribe(_ => {
-                    this.isLoaded$.next(true);
-                    this.currentSectionIndex = nextIndex;
-                    this.visitedSectionIndexes[nextIndex] = true;
-                    this.saveLastVisitedSectionIndex(nextIndex);
-                });
-            }
+            this.submitAnnouncementService.announceSubmit(null);
+            // The announcement could make listener components busy, but not instantly
+            // introduce a wait before we check whether we busy or not
+            timer(100).pipe(
```

Comment:
why it's 100? should we use this.timeToDebounce?

Reply:
Was trying to keep time to smallest possible value. The value of 250 would work too, I suppose.